collectors_app.php

```diff
@@ -21,68 +21,108 @@
-//----------------------------------------------------------------------------------------------------------------------preparing query
-$query = $db->prepare('SELECT * FROM `socks`');
-//----------------------------------------------------------------------------------------------------------------------execute query
-$result = $query->execute();
+// validation of inputs:
+// DB table for sizes, color, pattern ✅
+// dropdowns that populate colors, sizes, patterns ✅
 
-if ($result) {
-    $socks = $query->fetchAll();//--------------------------------------------------------------------------------------REASSIGN PLAYERS ARRAY
-} else {
-    echo 'not working';
+//----------------------------------------------------------------------------------------------------------------------dropdowns
+//----------------------------------------------------------------------------------------------------------------------connect DB function
+function connectDB() {
+    $db = new PDO(
+        'mysql:host=DB;dbname=collection',
+        'root',
+        'password'
+    );
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
 }
-//----------------------------------------------------------------------------------------------------------------------form $_GET data
-$test = '';
-if ($_GET) {
-    $test = "{$_GET['size']} {$_GET['pattern']} {$_GET['color']}";
-
-    //------------------------------------------------------------------------------------------------------------------move form to DB
-    $query = $db->prepare("INSERT INTO `socks` (`size`, `pattern`, `color`) VALUES (:size, :pattern, :color)");
-    $result = $query->execute([
-        'size' => $_GET['size'],
-        'pattern' => $_GET['pattern'],
-        'color' => $_GET['color']
-    ]);
+//----------------------------------------------------------------------------------------------------------------------get table function
+function getTable($table, $db) {
+    $query = $db->prepare("SELECT `{$table}` FROM `{$table}s`");
+    $result = $query->execute();
+    if ($result) {
+        return $query->fetchAll();
+    } else {
+        echo 'not working'; //maybe return not working
+    }
+}
+//----------------------------------------------------------------------------------------------------------------------creating html options for select dropdown
+function dropDownOptions($rows) {
+    $options= '';
+    foreach($rows as $row) {
+        foreach ($row as $key => $value) {
+            $options .= "<option value=\"{$value}\">{$value}</option>";
+        }
+    }
+    return $options;
 }
+//----------------------------------------------------------------------------------------------------------return full html dropdown - select, options, /select
+function createDropdown($name) {
 
+    $start = "<select name=\"{$name}\" id=\"{name}\"><option selected=\"selected\">$name</option>";
+    $options = dropDownOptions(getTable($name, connectDB()));
```

Comment:
Ok it's time to split this out a bit now. Calling layers of functions within functions is generally a bad idea.

It would be preferable to instead pass data in from outside using params.

Not just for best pratice reasons, in this case, you're creating lots of database connections when you only really need one.

Also, once that's done, please unit test this function